controllers/kafkacluster_controller.go

```diff
@@ -105,7 +106,15 @@ func (r *KafkaClusterReconciler) Reconcile(ctx context.Context, request ctrl.Req
 		return r.checkFinalizers(ctx, instance)
 	}
 
+	cruiseControlHealer, err := ccutil.NewCruiseControlHealer(ctx, instance)
+	if err != nil {
+		return requeueWithError(log, err.Error(), err)
+	}
+
 	if instance.Status.State != v1beta1.KafkaClusterRollingUpgrading {
+		if err := cruiseControlHealer.PauseSelfHealing(); err != nil {
```

Comment:
why are we pausing self healing here?
Or to better understand, in which states is the self healing paused?

Reply:
I agree this is a bit confusing. The reconcile function always transitions from running to reconciling to (maybe) upgrading. I'll split the pause logic from the CR state logic